templates/nvidia.yaml

```diff
@@ -0,0 +1,52 @@
+---
+apiVersion: kubeaddons.mesosphere.io/v1alpha1
+kind: Addon
+metadata:
+  name: nvidia
+  namespace: kubeaddons
+  labels:
+    kubeaddons.mesosphere.io/name: nvidia
+    kubeaddons.mesosphere.io/provides: nvidia
+  annotations:
+    appversion.kubeaddons.mesosphere.io/nvidia: "0.1"
+spec:
+  kubernetes:
+    minSupportedVersion: v1.15.0
+  namespace: kube-system
+  cloudProvider:
+    - name: aws
+      enabled: true
```

Comment:
@shaneutt will this enable it by default when deploying on AWS ? yes, right!

Reply:
From what perspective?

Reply:
Let's set `enabled: false` to be merge this PR asap. We won't enabled it until we get other changes in our master (kubeaddons controller)

Reply:
@hectorj2f i've just dismissed your review so this doesn't get accidentally merged until you're happy.

Reply:
@hectorj2f I can disable it be default.

However, FYI even if we enable the Nvidia Addon, daemonsets will not be installed unless GPU is enabled from cluster config:
```
  nodePools:
  - name: worker
    gpu:
      nvidia:
        enabled: false
  - name: gilbert
    gpu:
      nvidia:
        enabled: true
```

Because we have leveraged the nodeSelector and Konvoy has done the labeling part for GPU machines.

Reply:
We know that. I am asking to set enabled false here for aws just so we disabled it as an add-on to be attempted for deployment.

By the way the fact that Nvidia flag in cluster.yaml gets enabled, it doesn't mean this addon will be deployed. We need to handle that scenario in Konvoy or any component

Reply:
> By the way the fact that Nvidia flag in cluster.yaml gets enabled, it doesn't mean this addon will be deployed. We need to handle that scenario in Konvoy or any component

This is a good point. The addon depends on the cluster flag, and it does not mean the addon has to be deployed if the cluster flag is enabled. We can imagine a use case that a user expects that Konvoy handles the low-level container runtime (from the cluster flag) but they wanted to customize the nvidia driver and device plugin.

We don't necessarily need to enable the Nvidia addon by default!